### PR TITLE
Compression-based solving

### PR DESCRIPTION
**Pull request by aarmey**
This switches to performing a per-condition pre-compression step, then uses vanilla Pf2 from TensorLy. This is ultimately better because it is much faster, simpler, and will allow for datasets of virtually unlimited size. Some of the code here will ultimately be merged into TensorLy, and so disappear from here, but I don't see our solving changing much from here.

Note that the factors do solve with higher accuracy now. I'm not seeing the biology change, but they do shift around a bit.

Let me know as you encounter any issues using this. I'll merge as soon as you've both taken a look in case you have feedback.

### CODE REVIEW

File: sccp/imports.py [hunk 1]
```diff
@@ -7,19 +7,27 @@
...
+    _, sgIndex = np.unique(X.obs_vector(condition_name), return_inverse=True)
+    X.obs["condition_unique_idxs"] = sgIndex
+
+    # Pre-calculate gene means
+    means, _ = mean_variance_axis(X.X, axis=0)  # type: ignore
+    X.var["means"] = means
+
     # Filter out genes with too few reads
-    X = X[:, readSum > 1000]
+    X = X[:, readSum > 100]
```
Comment (aarmey) [inline sccp/imports.py:30]: Lowered the threshold for including a gene. This includes many more genes within the Thomson study.

File: sccp/imports.py [hunk 2]
```diff
@@ -7,19 +7,27 @@
...
-    readSum *= X.X.shape[0]
-    inplace_column_scale(X.X, 1.0 / readSum)
+    readSum *= X.shape[0]
+    inplace_column_scale(X.X, 1.0 / (readSum + 1.0e-9))
 
     # Transform values
     X.X.data = np.log10((1000.0 * X.X.data) + 1.0)  # type: ignore
 
+    # Get the indices for subsetting the data
+    _, sgIndex = np.unique(X.obs_vector(condition_name), return_inverse=True)
```
Comment (aarmey) [inline sccp/imports.py:22]: The condition labels are pre-calculated during loading, so this isn't done repeatedly later.

**Comment by andrewram4287** [comment]
@aarmey no issues running this new version with the thomson dataset, but the reconstruction error is fairly highly now, this reconstruction is regarding the decomposition of the "cell state" tensor correct?

**Comment by aarmey** [comment]
@andrewram4287 you can ignore the reconstruction error printed out by TensorLy—it's actually wrong and needs to be fixed within the package.

File: sccp/imports.py (hunk 1 above)
Comment (aarmey) [reply sccp/imports.py:30]: Yes. Previously we were removing more.

**Review by andrewram4287 — APPROVED** [review]

**Review by borcuttjahns — APPROVED** [review]

**Pull request merged**